docs/csharp/tutorials/string-interpolation.md

```diff
@@ -1,122 +1,98 @@
+You specify a format string that is supported by the type of the expression result by following the interpolated expression with a colon (":") and the format string:
 
 ```
-dotnet new console
+{<interpolatedExpression>:<formatString>}
 ```
 
-This command creates a barebones .NET Core project with a project file, *interpolated.csproj*, and a source code file, *Program.cs*. You will need to execute `dotnet restore` to restore the dependencies needed to compile this project.
+The following example shows how to specify standard and custom format strings for expressions that produce date and time or numeric results:
 
-[!INCLUDE[DotNet Restore Note](~/includes/dotnet-restore-note.md)]
+[!code-csharp-interactive[format string example](~/samples/snippets/csharp/tutorials/string-interpolation/Program.cs#2)]
 
-To execute the program, use `dotnet run`. You should see "Hello, World" output to the console.
+For more information, see the [Format String Component](../../standard/base-types/composite-formatting.md#format-string-component) section of the [Composite Formatting](../../standard/base-types/composite-formatting.md) topic. That section provides links to the topics that describe standard and custom format strings supported by .NET base types.
 
+## How to control the field width and alignment of the formatted interpolated expression
 
+You specify the minimum field width in which string representation of the expression result is inserted and the text alignment within that field by following the interpolated expression with a comma (",") and the constant expression that defines alignment:
 
-## Intro to String Interpolation
+```
+{<interpolatedExpression>,<alignment>}
+```
 
-With <xref:System.String.Format%2A?displayProperty=nameWithType>, you specify "placeholders" in a string that are replaced by the arguments following the string. For instance:
+If the alignment value is positive, the formatted string representation of the expression result is right-aligned; if negative, it's left-aligned.
 
-[!code-csharp[String.Format example](../../../samples/snippets/csharp/new-in-6/string-interpolation.cs#StringFormatExample)]  
+If you need to specify both alignment and a format string, start with the alignment component:
 
-That will output "My name is Matt Groves".
+```
+{<interpolatedExpression>,<alignment>:<formatString>}
+```
 
-In C# 6, instead of using `String.Format`, you define an interpolated string by prepending it with the `$` symbol, and then using the variables directly in the string. For instance:
+The following example shows how to specify alignment:
 
-[!code-csharp[Interpolation example](../../../samples/snippets/csharp/new-in-6/string-interpolation.cs#InterpolationExample)]  
+[!code-csharp-interactive[alignment example](~/samples/snippets/csharp/tutorials/string-interpolation/Program.cs#3)]
```

Comment:
- I'm not sure what the "|" characters accomplish. (Were you thinking of a markdown table?) I think that I'd remove them
- NameAlignment should be 10 or greater, since "Arithmetic" has 10 characters.

Reply:
I find pipes ("|") in output make it easier to visualize alignment. For example, see the [current version of the tutorial](https://docs.microsoft.com/en-us/dotnet/csharp/tutorials/string-interpolation#how-string-interpolation-works), the example with two numeric columns: from the code, the right column is aligned to the right, while from output, it looks as if it's left-aligned. Just not a good choice of the field width makes the example confusing. With the pipes in output example would be clearer.

As for the `NameAlignment` value, it's an intentional choice in order to show that the *minimum* field width is specified and what happens if the formatted text is longer.

Reply:
In that case, it would be good to explain the pipes. Somehow with pipes in text, I (at least) expect proper alignment.